utils/HtmlElems.ts

```diff
@@ -0,0 +1,183 @@
+${content.join( "" )}
+</div>`
+
+// Div with custom class
+export const DivWithClass = ( divClass: string, ...content: Array<string> ) => `<div class=${divClass}>
+            ${content.join( "" )}
+</div>`
+
+
+export const SearchBtn = ( clickedBtn: CurrentButton ) => {
+    let style: string
+    if ( clickedBtn === CurrentButton.Search ) {
+        style = "opacity: 100%"
+    } else {
+        style = "opacity:0.5"
+    }
+    return `<a id="searchBtn"
+    class="waves-effect  waves-light btn" style=${style}>
+    Search
+</a>`
+}
+
+export const AddBtn = ( clickedBtn: CurrentButton ) => {
+    let style: string
+    if ( clickedBtn === CurrentButton.Add ) {
+        style = "opacity: 100%"
+    } else {
+        style = "opacity:0.5"
+    }
+    return `<a id="addBtn"
+    class="waves-effect  waves-light btn" style=${style}>
+    Add
+</a>`
+}
+
+
+export const RemoveBtn = ( clickedBtn: CurrentButton ) => {
+    let style: string
+    if ( clickedBtn === CurrentButton.Remove ) {
+        style = "opacity: 100%"
+    } else {
+        style = "opacity:0.5"
+    }
+    return `<a id="removeBtn"
+    class="waves-effect  waves-light btn" style=${style}>
+    Remove
+</a>`
+}
+export const ShowAllBtn = ( clickedBtn: CurrentButton ) => {
+    let style: string
+    if ( clickedBtn === CurrentButton.ShowAll ) {
+        style = "opacity: 100%"
+    } else {
+        style = "opacity:0.5"
+    }
+    return `<a id="showAllBtn"
+    class="waves-effect  waves-light btn" style=${style}>
+    Show All
+</a>`
+
+}
+
+
+// Submit button as html
+export const SubmBtn = ( clickedBtn: CurrentButton ) => {
+    let visability: string
+    let btnName: string
+    if ( clickedBtn === CurrentButton.Search ) {
+        visability = "block"
+        btnName = "Search"
+    }
+    if ( clickedBtn === CurrentButton.Add ) {
+        visability = "block"
+        btnName = "Add"
+    }
+    if ( clickedBtn === CurrentButton.Remove ) {
+        visability = "block"
+        btnName = "Remove"
+    }
+    if ( clickedBtn === CurrentButton.ShowAll )
+        visability = "none"
+
+    return `<a id="submitBtn" ${status}
+    class="waves-effect  waves-light btn red" style="display: ${visability};" >
+   ${btnName}
+    </a>`
+
+}
+
+
+// Space between elements 
+export const Space = () => `<br>
+                            <br>
+                            <br>`
+
+// Hidden by default div 
+export const HiddenDiv = ( clickedBtn: CurrentButton, users: Array<User>, ifSbmtClicked: boolean, ...content: Array<string> ) => {
+    let status: string
+    if ( clickedBtn === CurrentButton.Search )
+        status = "hidden"
+    if ( clickedBtn === CurrentButton.Search, users.length > 0, ifSbmtClicked === true )
+        status = "block"
+    if ( clickedBtn === CurrentButton.Add )
+        status = "hidden"
+    if ( clickedBtn === CurrentButton.Remove )
+        status = "hidden"
+    if ( clickedBtn === CurrentButton.ShowAll )
+        status = "block"
```

Comment:
These should be else-ifs, otherwise every time it checks every single condition before returning.

OR, you could use `switch` on clickedBtn. Would be much cleaner.